rest/route/host_agent.go

```diff
@@ -663,6 +683,15 @@ func assignNextAvailableTask(ctx context.Context, env evergreen.Environment, tas
 			continue
 		}
 
+		err = task.SetNumNextTaskDispatches(nextTask.Id, nextTask.Execution, nextTask.NumNextTaskDispatches+1)
+		if err != nil {
```

Comment:
nit: L687-692 can be simplified down to `grip.Error(message.WrapError(err, message.Fields { ...}))` because `message.WrapError` already suppresses logging for nil errors.